File: font/math.js
/**
 * Returns the weighted average of two values.
 *
 * @param {number} z
 * @param {number} w
 * @param {number} ratio The weighting factor of z
 * @returns {number} The weighted average of z and w
 */
export function mix(z, w, ratio) {
  return w + ratio * (z - w)
}

File: crates/monoxide-font/src/math.rs
use std::ops::{Add, Mul, Sub};

/// Returns the weighted average of `z` and `w` based on the `ratio`.
/// The weight is put on `z`, i.e. it returns `w + (z - w) * ratio`.
pub fn mix<F, T, D>(z: T, w: T, ratio: F) -> T
where
    T: Sub<T, Output = D> + Copy,
    D: Mul<F>,
    T: Add<<D as Mul<F>>::Output, Output = T>,
{
    w + (z - w) * ratio
}
